### PR TITLE
[SKU Scanner] Events Tracking

### LINKED ISSUE #9660: [SKU Scanner] Track events
- Events: pdfdoF-2Ql#comment-4137-p2
- Add `barcode_format` property when fails to scan, reason: pdfdoF-2Zq-p2

### PR DESCRIPTION
**Pull request by toupper**
Closes: #9660 

## Description

With this PR we implement the events tracking for everything related to the Scanned and Deliver Project. As our current solution never gives an error from the barcode scanner itself, we do not track the `barcode_scanning_failure` event.

### Tap on the barcode button in the Order list
Go to the Orders tab and tap on the barcode button in the left corner:

`🔵 Tracked order_list_product_barcode_scanning_tapped, properties: [AnyHashable("blog_id"): 214354650, AnyHashable("is_wpcom_store"): true]`

### Tap on the barcode button in Order Creation

`🔵 Tracked order_creation_product_barcode_scanning_tapped, properties: [AnyHashable("is_wpcom_store"): true, AnyHashable("blog_id"): 214354650]`

#### On Order List

`🔵 Tracked barcode_scanning_success, properties: [AnyHashable("blog_id"): 214354650, AnyHashable("is_wpcom_store"): true, AnyHashable("source"): "order_list"]`

#### On Order Creation

`🔵 Tracked barcode_scanning_success, properties: [AnyHashable("source"): "order_creation", AnyHashable("is_wpcom_store"): true, AnyHashable("blog_id"): 214354650]`

#### On Order List

`🔵 Tracked product_search_via_sku_success, properties: [AnyHashable("blog_id"): 214354650, AnyHashable("source"): "order_list", AnyHashable("is_wpcom_store"): true]
`
#### On Order Creation

`🔵 Tracked product_search_via_sku_success, properties: [AnyHashable("blog_id"): 214354650, AnyHashable("source"): "order_creation", AnyHashable("is_wpcom_store"): true]`

#### On Order List

`🔵 Tracked product_search_via_sku_failure, properties: [AnyHashable("barcode_format"): "VNBarcodeSymbologyEAN13", AnyHashable("reason"): "The operation couldn’t be completed. (Yosemite.ProductLoadError error 1.)", AnyHashable("source"): "order_list", AnyHashable("is_wpcom_store"): true, AnyHashable("blog_id"): 214354650]`

#### On Order Creation

`🔵 Tracked product_search_via_sku_failure, properties: [AnyHashable("barcode_format"): "VNBarcodeSymbologyEAN13", AnyHashable("reason"): "The operation couldn’t be completed. (Yosemite.ProductLoadError error 1.)", AnyHashable("source"): "order_creation", AnyHashable("is_wpcom_store"): true, AnyHashable("blog_id"): 214354650]`

#### After scanning a barcode in the order list

`🔵 Tracked order_product_add, properties: [AnyHashable("blog_id"): 214354650, AnyHashable("added_via"): "scanning", AnyHashable("source"): "order_list", AnyHashable("flow"): "creation", AnyHashable("is_wpcom_store"): true, AnyHashable("product_count"): 1]`

#### After scanning a barcode in the order creation screen

`Tracked order_product_add, properties: [AnyHashable("flow"): "creation", AnyHashable("blog_id"): 214354650, AnyHashable("product_count"): 1, AnyHashable("source"): "order_creation", AnyHashable("added_via"): "scanning", AnyHashable("is_wpcom_store"): true]`

#### After adding products from the product selector

`🔵 Tracked order_product_add, properties: [AnyHashable("blog_id"): 214354650, AnyHashable("added_via"): "manually", AnyHashable("product_count"): 2, AnyHashable("source"): "order_creation", AnyHashable("is_wpcom_store"): true, AnyHashable("flow"): "creation"]`

## Screenshots

---
- [X] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/Classes/ViewRelated/Orders/OrdersRootViewController.swift [hunk 1]
```diff
@@ -203,18 +203,25 @@ final class OrdersRootViewController: UIViewController {
...
     @objc func presentOrderCreationFlowByProductScanning() {
+        ServiceLocator.analytics.track(event: WooAnalyticsEvent.Orders.orderAddNewFromBarcodeScanningTapped())
+
         guard let navigationController = navigationController else {
             return
         }
 
         let productSKUBarcodeScannerCoordinator = ProductSKUBarcodeScannerCoordinator(sourceNavigationController: navigationController,
                                                                                       onSKUBarcodeScanned: { [weak self] scannedBarcode in
+            ServiceLocator.analytics.track(event: WooAnalyticsEvent.Orders.barcodeScanningSuccess(from: .orderList))
```
Comment (iamgabrielma) [inline WooCommerce/Classes/ViewRelated/Orders/OrdersRootViewController.swift]:
```suggestion
            analytics.track(event: WooAnalyticsEvent.Orders.barcodeScanningSuccess(from: .orderList))
```

We have a reference to `ServiceLocator.analytics` in the class declaration, we can remove this one and reference the existing one. Added to this, perhaps we should inject it as well in the init?

File: WooCommerce/Classes/ViewRelated/Orders/OrdersRootViewController.swift [hunk 2]
```diff
@@ -203,18 +203,25 @@ final class OrdersRootViewController: UIViewController {
     /// Presents the Order Creation flow when a product is scanned
     ///
     @objc func presentOrderCreationFlowByProductScanning() {
+        ServiceLocator.analytics.track(event: WooAnalyticsEvent.Orders.orderAddNewFromBarcodeScanningTapped())
```
Comment (iamgabrielma) [inline WooCommerce/Classes/ViewRelated/Orders/OrdersRootViewController.swift]: Same as below: 

```suggestion
        analytics.track(event: WooAnalyticsEvent.Orders.orderAddNewFromBarcodeScanningTapped())
```

**Review by iamgabrielma — APPROVED** [review]
Tests well!

✅ Tap on the barcode button in the Order list
✅ Tap on the barcode button in Order Creation
✅ Scan a barcode successfully
✅ Find a product successfully after the SKU Search
✅ Finding a product returns an error after the SKU Search
✅ Adding Product

I left a comment regarding using existing analytics property rather than retrieving from ServiceLocator, but not a blocker 🚢

File: WooCommerce/Classes/ViewRelated/Orders/OrdersRootViewController.swift (hunk 1 above)
Comment (toupper) [reply WooCommerce/Classes/ViewRelated/Orders/OrdersRootViewController.swift]: Thanks for the comment! Indeed, it makes sense to change it, I did it in 13fa2b7ddc685c707e8979a28c1b73787d0dc996 

> Added to this, perhaps we should inject it as well in the init?

I think for now as we don't have tests for this view controller we can keep it without being injected, but if the requirements change we can do it 👍

**Pull request merged**